src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Linux.cs

```diff
@@ -78,33 +78,10 @@ internal static DateTime BootTimeToDateTime(TimeSpan timespanAfterBoot)
             return dt.ToLocalTime();
         }
 
+        private static DateTime _bootTime;
         /// <summary>Gets the system boot time.</summary>
-        private static DateTime BootTime
-        {
-            get
-            {
-                // '/proc/stat -> btime' gets the boot time.
-                // btime is the time of system boot in seconds since the Unix epoch.
-                // It includes suspended time and is updated based on the system time (settimeofday).
-                const string StatFile = Interop.procfs.ProcStatFilePath;
-                string text = File.ReadAllText(StatFile);
-                int btimeLineStart = text.IndexOf("\nbtime ", StringComparison.Ordinal);
-                if (btimeLineStart >= 0)
-                {
-                    int btimeStart = btimeLineStart + "\nbtime ".Length;
-                    int btimeEnd = text.IndexOf('\n', btimeStart);
-                    if (btimeEnd > btimeStart)
-                    {
-                        if (long.TryParse(text.AsSpan(btimeStart, btimeEnd - btimeStart), out long bootTimeSeconds))
-                        {
-                            return DateTime.UnixEpoch + TimeSpan.FromSeconds(bootTimeSeconds);
-                        }
-                    }
-                }
-
-                return DateTime.UtcNow;
-            }
-        }
+        private static DateTime BootTime => _bootTime != default ? _bootTime :
```

Comment:
This is still not guaranteed to be atomic on 32-bit systems.

Reply:
Ah, right. Another option (to keep it lock-free) is to have a boolean flag:

```c#
        private static bool s_initBootTime;
        private static DateTime _bootTime;
        /// <summary>Gets the system boot time.</summary>
        private static DateTime BootTime 
        {
            get
            { 
               if (!s_initBootTime)
               {
                   _bootTime = DateTimeOffset.FromUnixTimeMilliseconds(InteropSysGetBootTimeMilliseconds()).UtcDateTime;
                   s_initBootTime = true;
               }

               return _bootTime;
           }
       }
```
I didn't cached it at first. During the review it was pointed out that value should remain unchanged across the accesses. We can also cache its value on the native side.

Reply:
I think it would be easiest to cache the ticks as `long`. You can use `Interlocked.CompareExchange` for that. It will guarantee that you are caching the value exactly once and atomically. Constructing DateTime from ticks is cheap.